File: Checks/PWR031/solution.f90
! PWR031: Replace pow by multiplication, division and/or square root

program main
  use iso_fortran_env, only : real64
  implicit none
  !
  print '(A, F0.15)', '2 raised to the power of 1.5 is: ', &
      raise_x_to_the_power_of_1_point_5(2.0_real64)
contains
  pure function raise_x_to_the_power_of_1_point_5(x)
    implicit none
    ! function return type
    real(kind=real64) :: raise_x_to_the_power_of_1_point_5
    ! dummy args
    real(kind=real64), intent(in) :: x
    !
    raise_x_to_the_power_of_1_point_5 = x * sqrt(x)
  end function raise_x_to_the_power_of_1_point_5
end program main

File: Checks/PWR031/solution.c
// PWR031: Replace pow by multiplication, division and/or square root

#include <math.h>
#include <stdio.h>

__attribute__((const)) double raise_x_to_the_power_of_1_point_5(double x) {
  return x * sqrt(x);
}

int main() {
  printf("2 raised to the power of 1.5 is: %0.15f\n",
         raise_x_to_the_power_of_1_point_5(2.0));
  return 0;
}
